### DISCUSSION TITLE
issue with array initialization using index

### DISCUSSION THREAD

**Discussion by Lord-Vlad** [vlang/v#23224] [Questions and Answers] [answered]
I just started working with V, and while experimenting with arrays I stumbled upon the following.
When you initialize an `i32` array, you can't use index directly, you have to cast to `i32`.
Aren't `int` and `i32` the same thing?
This also applies to unsigned numbers, although I suspect it's due to the lack of promotion? Floating point numbers do work.

```v
array := []i32{len: 3, init: index}

error: expected `i32`, not `int`
```

**Comment by jorgeluismireles** [comment]
Read this section: https://docs.vlang.io/v-types.html#primitive-types

**Comment by JalonSolov** [comment, marked as answer]
For now, `i32` and `int` _should_ be treated the same.  `int` is supposed to be changed at some point so it will either be `i32` or `i64`, depending on the system.

In any case, this is a bug.  `index` should always be the correct type to init the array, not a fixed `int`.

https://github.com/vlang/v/issues/23225
